### DISCUSSION TITLE
EmailJS React Email send working on Google Chrome but not Safari

### DISCUSSION THREAD

**Discussion by sanjaynatesan** [emailjs-com/help#16] [Q&A]
Title... + For Safari the console does not give a straightforward error, just prints Failed... and empty string for the error. Here is my code below (IDs redacted for privacy), where params are the list of parameters I'm trying to send to the service... any ideas why this is working on one browser but not the other? Also, this is a deployed service, using Heroku, but I'm not getting any indications that its an error on that side.

`emailjs
        .send(
          'REMOVED',
          'REMOVED',
          params,
          'REMOVED'
        )
        .then(
          function (response) {
            console.log('SUCCESS!', response.status, response.text);
          },
          function (error) {
            console.log('FAILED...', error);
          }
        );`

**Comment by xr0master** [comment]
Hi @sanjaynatesan 
The issue is not with the EmailJS service itself, but with how the code has been created to call the function. Both browsers handle events slightly differently. Can you provide more code?

**Reply by sanjaynatesan** [reply]
Sure, for some background on the page load there is a conditional, and based on the result of it two emails will be sent (the conditional will determine the parameters of the email)

Also, here is a larger portion of the code. Please bear in mind that a large amount of the code was removed as it does not pertain to the EmailJS service or the creation of the parameters for it. Further, the PDF generated using jsPDF was visible on the screen, so I don't believe the issue has to do with that side of it. Someone has previously mentioned using the preventDefault method, but I am extremely confused on how I would implement this in my case.

```js
  const len = answers.length;
  const [buttonclick, setButton] = useState(false);
  var sentEmail = 0;
  if (len === 2) {
    const pdfData = doc.output('datauristring'); //some jsPDF document that was generated to be sent over the email
    const pdfDatauri = doc.output('datauri');

    if (sentEmail === 0) {
      sentEmail = 1;
      var params = {
        SOME PARAMS (includes URI for pdf created using jsPDF),
      };
      emailjs
        .send(
          REMOVED,
          REMOVED,
          params,
          REMOVED
        )
        .then(
          function (response) {
            console.log('SUCCESS!', response.status, response.text);
          },
          function (error) {
            console.log('FAILED...', error);
          }
        );
      if (emailaddress !== '') {
        params = {
          SOME PARAMS (includes URI for pdf created using jsPDF),
        };
        emailjs
          .send(
            REMOVED,
            REMOVED,
            params,
            REMOVED
          )
          .then(
            function (response) {
              console.log('SUCCESS!', response.status, response.text);
            },
            function (error) {
              console.log('FAILED...', error);
            }
          );
      }
    }

    return (
      {\*some different visible movement that doesn’t involve the emailjs path */}
         );
  } else if (len === 4) {
    const pdfData = doc.output('datauristring'); //some jsPDF document that was generated to be sent over the email
    const pdfDatauri = doc.output('datauri');

    if (sentEmail === 0) {
      sentEmail = 1;
      var params = {
        SOME DIFFERENT PARAMS (includes URI for pdf created using jsPDF),
      };
      emailjs
        .send(
          REMOVED,
          REMOVED,
          params,
          REMOVED
        )
        .then(
          function (response) {
            console.log('SUCCESS!', response.status, response.text);
          },
          function (error) {
            console.log('FAILED...', error);
          }
        );
      if (emailaddress !== '') {
        params = {
          SOME DIFFERENT PARAMS (includes URI for pdf created using jsPDF),
        };
        emailjs
          .send(
            REMOVED,
            REMOVED,
            params,
            REMOVED
          )
          .then(
            function (response) {
              console.log('SUCCESS!', response.status, response.text);
            },
            function (error) {
              console.log('FAILED...', error);
            }
          );
      }
    }
    return (
      {\*some different visible movement that doesn’t involve the emailjs path */}
    );
  }
}

export default ResultsPage;

```

**Reply by xr0master** [reply]
As far as I understand, the code is immediately run as soon as the ResultsPage component is called. I suggest wrapping your code with a `useEffect` hook to call it when the component mounts.
